library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -1,76 +1,75 @@
 package com.novoda.downloadmanager;
 
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
 import android.util.Log;
 
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 
-import static com.novoda.downloadmanager.DownloadBatchStatus.Status;
-
-class VersionOneToVersionTwoMigrator implements Migrator {
+class MigrationJob implements Runnable {
 
     private static final String TAG = "V1 to V2 migrator";
 
     private static final int RANDOMLY_CHOSEN_BUFFER_SIZE_THAT_SEEMS_TO_WORK = 4096;
     private static final String TABLE_BATCHES = "batches";
     private static final String WHERE_CLAUSE_ID = "_id = ?";
 
-    private final MigrationExtractor migrationExtractor;
-    private final DownloadsPersistence downloadsPersistence;
-    private final InternalFilePersistence internalFilePersistence;
-    private final SqlDatabaseWrapper database;
-    private final UnlinkedDataRemover unlinkedDataRemover;
-    private final MigrationService migrationService;
-    private final NotificationChannelCreator channelCreator;
-    private final NotificationCreator<MigrationStatus> notificationCreator;
-    private final InternalMigrationStatus internalMigrationStatus;
-
-    VersionOneToVersionTwoMigrator(MigrationExtractor migrationExtractor,
-                                   DownloadsPersistence downloadsPersistence,
-                                   InternalFilePersistence internalFilePersistence,
-                                   SqlDatabaseWrapper database,
-                                   UnlinkedDataRemover unlinkedDataRemover,
-                                   MigrationService migrationService,
-                                   NotificationChannelCreator channelCreator,
-                                   NotificationCreator<MigrationStatus> notificationCreator,
-                                   InternalMigrationStatus internalMigrationStatus) {
-        this.migrationExtractor = migrationExtractor;
-        this.downloadsPersistence = downloadsPersistence;
-        this.internalFilePersistence = internalFilePersistence;
-        this.database = database;
-        this.unlinkedDataRemover = unlinkedDataRemover;
-        this.migrationService = migrationService;
-        this.channelCreator = channelCreator;
-        this.notificationCreator = notificationCreator;
-        this.internalMigrationStatus = internalMigrationStatus;
+    private final Context context;
+    private final File databasePath;
+    private final List<MigrationCallback> migrationCallbacks = new ArrayList<>();
+
+    MigrationJob(Context context, File databasePath) {
+        this.context = context;
+        this.databasePath = databasePath;
+    }
+
+    void addCallback(MigrationCallback callback) {
+        migrationCallbacks.add(callback);
     }
 
-    @Override
-    public void migrate() {
+    public void run() {
```

Comment:
Each job is now self contained and creates all of it's dependencies when running. We now return a `DB_NOT_PRESENT` if the file doesn't exist rather than just terminating 😬